pkg/client/cache/index.go

```diff
@@ -28,6 +28,8 @@ type Indexer interface {
 	Store
 	// Retrieve list of objects that match on the named indexing function
 	Index(indexName string, obj interface{}) ([]interface{}, error)
+	// ListIndexKeys returns the list of keys for an index
```

Comment:
I know I originally suggested it, but I think we should call this something different to avoid any confusion internally with the ```Store.ListKeys``` method that returns the ```KeyFunc``` value because I could see how someone may one day take the result from this and pass it into ```GetByKey```.

In practice, we are trying to get the unique set of values that are mapped against a particular ```IndexFunc```.

How about the following:

```
// ListIndexFuncValues returns the list of unique values from the named indexing function.
ListIndexFuncValues(indexName string) []string
```